src/compiler/checker.ts

```diff
@@ -7283,7 +7285,7 @@ module ts {
                 case SyntaxKind.TypeAssertionExpression:
                     return checkTypeAssertion(<TypeAssertion>node);
                 case SyntaxKind.ParenthesizedExpression:
-                    return checkExpression((<ParenthesizedExpression>node).expression);
+                    return checkExpression((<ParenthesizedExpression>node).expression, contextualMapper);

```

Comment:
contextualTypingWithFixedTypeParameters1.ts already tests it. The error message about a 'foo' property missing on type 'T' reveals the problem.